organisaatio-service/src/main/java/fi/vm/sade/organisaatio/service/filters/CsrfHeaderFilter.java

```diff
@@ -0,0 +1,44 @@
+package fi.vm.sade.organisaatio.service.filters;
+
+import javax.servlet.*;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.util.Enumeration;
+
+public class CsrfHeaderFilter implements Filter {
+
+    private final String CSRF_HEADER_NAME = "CSRF";
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+        CsrfServletRequestWrapper csrfServletRequestWrapper = new CsrfServletRequestWrapper(httpServletRequest);
+
+        Enumeration<String> csrfHeader = httpServletRequest.getHeaders(CSRF_HEADER_NAME);
+        if (csrfHeader.hasMoreElements() || httpServletRequest.getCookies() == null) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        for (Cookie cookie : httpServletRequest.getCookies()) {
```

Comment:
Jos tämän laittaa if sisään ylemmän sijaan voi filterChain.doFilter(request, response); ja return; jättää kummassakin blokissa pois.

Reply:
Silloin kun headerin lisääminen onnistuu, on tarkoitus käyttää request wrapperia, muuten käytetään tavallista requestia

Reply:
Sen voi mielestäni laittaa tuohon samaan request/httpServletRequest muuttujaan joka lopuksi syötettän doFilteriin(). Tämän sijoituksen voi laittaa if()for()if()break; sisään. Mutta tämä vain refaktorointi-ideana.